crates/parser/src/format/parquet.rs

```diff
@@ -0,0 +1,103 @@
+//! Parser for the parquet format. This will accept any stream of parquet files, limited up to 1GB
+//! in size.
+use super::{Input, Output, ParseError, Parser};
+use parquet::file::reader::{FileReader, SerializedFileReader};
+use parquet::record::reader::RowIter;
+use std::convert::TryFrom;
+use serde_json::Value;
+
+struct ParquetParser;
+
+pub fn new_parser() -> Box<dyn Parser> {
+    Box::new(ParquetParser)
+}
+
+const MAX_FILE_SIZE: i64 = 1024 * 1024 * 1024;
```

Comment:
I'd expect us to be able to read a file row group by row group, and if there was going to a be limit it would be based on the row group size and not the total file size.

I'm also wondering how this works with compressed files - is the metadata byte size for compressed data or uncompressed data? If `rg.total_byte_size()` is telling us about compressed size it may be dramatically different when uncompressed and loaded into memory. But I am not sure about this.

Reply:
@williamhbaker this is the uncompressed size. Good point about row groups, I'm going apply the limit on a row group level rather than the whole file size... that is after verifying that's how the data is read 🤔